syntax = "proto3";

/* At the moment is only being requested the review by ID and All the reviews of a game. The other services will implemented after checking if it works. */


// ######################
// ###### REQUESTS ###### 
// ######################

// Search a review by ID
message ReviewByIdRequest {
	string review_id = 1;
}

// Search for reviews by game
message ReviewsByGameRequest {
    string app_id = 1;
	int32 max_results = 2;
}


// ######################
// ##### RESPONSES ###### 
// ######################

message ReviewData {
	string app_id = 1;
	string review_id = 2;
	string language = 3;
	string review = 4;
	string timestamp_created = 5;
	string timestamp_updated = 6;
	string recommended = 7;
	string votes_helpful = 8;
	string author_steam_id = 9;
}

message ReviewDataResponse {
    repeated ReviewData reviews = 1;
}


// ######################
// ##### SERVICES ###### 
// ######################

// Services provided by the reviews microservice
service Reviews {
    rpc GetReview (ReviewByIdRequest) returns (ReviewData);
	rpc GetGameReviews (ReviewsByGameRequest) returns (ReviewDataResponse);
}
